modules/auxiliary/server/dns/native_server.rb

```diff
@@ -37,11 +46,13 @@ def run
       service.wait
     rescue Rex::BindFailed => e
       print_error "Failed to bind to port #{datastore['RPORT']}: #{e.message}"
-    ensure
-      stop_service(true)
     end
   end
 
+  def cleanup
+    stop_service(true)
```

Comment:
Will this have trouble with `cleanup_service` name change in #16316

Reply:
Good catch; there will almost certainly be a conflict between them upon merging the second one in. Maybe we press pause on this one and I can rebase once that one lands?

Reply:
Well that PR has been landed so it looks like we'll be ready to move forward with this one once it's been rebased :+1: 

Thanks @smashery !

Reply:
Rebasing done.